praw/models/reddit/subreddit.py

```diff
@@ -785,32 +803,16 @@ def __init__(self, subreddit):
         """
         self.subreddit = subreddit
 
-    def __iter__(self):
-        """Iterate through the user flair templates.
-
-        Example:
-
-        .. code-block:: python
-
-           for template in reddit.subreddit('NAME').flair.templates:
-               print(template)
-
-
-        """
-        url = API_PATH['flairselector'].format(subreddit=self.subreddit)
-        data = {'unique': self.subreddit._reddit._next_unique}
-        for template in self.subreddit._reddit.post(url, data=data)['choices']:
-            yield template
-
-    def add(self, text, css_class='', text_editable=False, is_link=False):
+    def add(self, text, css_class='', text_editable=False, is_link=None):
         """Add a flair template to the associated subreddit.
 
         :param text: The flair template's text (required).
         :param css_class: The flair template's css_class (default: '').
         :param text_editable: (boolean) Indicate if the flair text can be
             modified for each Redditor that sets it (default: False).
-        :param is_link: (boolean) When True, add a link flair template rather
-            than a Redditor flair template (default: False).
+        :param is_link: (boolean) When None, choose a template type in a
+            subclass-defined way. Otherwise, add a link flair template when
+            True and a Redditor flair template when false (default: None).
```

Comment:
This docstring has the potential to be fairly confusing. Without the latest commit, we could just have separate docstrings in different subclasses, but at the cost of a lot of duplication.